File: FSharp/GenMonad1/Program.fs
open FsCheck
open Gen
open System

[<EntryPoint>]
let main _ = 

    let g1 = gen {
        let! s = Arb.generate<string> |> suchThat (String.IsNullOrEmpty >> not)
        let! c = elements s
        return (s, c)
    }
    sample 50 10 g1 |> Seq.iter (printfn "%A")

    printfn "%s" (new string('-', 80))

    let g2 = gen {
        let! s = Arb.generate<string> |> suchThat (String.IsNullOrEmpty >> not)
        return! elements s |> map (fun c -> (s, c))
    }
    sample 50 10 g2 |> Seq.iter (printfn "%A")

    0

File: CSharp/GenMonad1/Program.cs
using System;
using System.Linq;
using FsCheck;
using FsCheck.Fluent;

namespace GenMonad1
{
    internal static class Program
    {
        private static void Main()
        {
            var g1 =
                from s in Arb.generate<string>()
                where !string.IsNullOrEmpty(s)
                from c in Gen.elements(s)
                select Tuple.Create(s, c);

            foreach (var sample in Gen.sample(10, 10, g1)) Console.WriteLine(sample);

            Console.WriteLine(new string('-', 80));

            var g2 =
                from s in Any.OfType<string>()
                where !string.IsNullOrEmpty(s)
                from c in Any.ValueIn(s.AsEnumerable())
                select Tuple.Create(s, c);

            foreach (var sample in Gen.sample(10, 10, g2)) Console.WriteLine(sample);
        }
    }
}
